src/Microsoft.ML.Data/Transforms/NormalizeTransform.cs

```diff
@@ -197,6 +197,39 @@ private static VersionInfo GetVersionInfo()
             SetMetadata();
         }
 
+        /// <summary>
+        /// Potentially apply a min-max normalizer to the data's feature column, keeping all existing role
+        /// mappings except for the feature role mapping.
+        /// </summary>
+        /// <param name="env">The host environment to use to potentially instantiate the transform</param>
+        /// <param name="data">The role-mapped data that is potentially going to be modified by this method.</param>
+        /// <param name="trainer">The trainer to query with <see cref="NormalizeUtils.NeedNormalization(ITrainer)"/>.
+        /// This method will not modify <paramref name="data"/> if the return from that is <c>null</c> or
+        /// <c>false</c>.</param>
+        /// <returns>True if the normalizer was applied and <paramref name="data"/> was modified</returns>
+        public static bool CreateIfNeeded(IHostEnvironment env, ref RoleMappedData data, ITrainer trainer)
```

Comment:
Great! I will be using this method in #424 just before calling train method, correct?

Reply:
I think so. Also probably before the hypothetical cache-if-needed.